src/App.tsx

```diff
@@ -9,37 +9,81 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { User } from './types/User';
+import { Post } from './types/Post';
+
+import { getPosts } from './api/ApiMethods';
+
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isPostOpen, setIsPostOpen] = useState<Post | null>(null);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setLoading(true);
+      setHasError(false);
+
+      getPosts(selectedUser.id)
+        .then(posts => setUserPosts(posts))
```

Comment:
```suggestion
        .then(setUserPosts)
```
we can simplify this logic